src/map_io/map_buildingdata_packet.cc

```diff
@@ -807,6 +807,9 @@ void MapBuildingdataPacket::read_productionsite(ProductionSite& productionsite,
 				}
 			}
 
+			// TODO(Nordfriese): Savegame compatibility
+			productionsite.infinite_production_ = packet_version >= 10 && fr.unsigned_8() > 0;
```

Comment:
this is dependent on short circuit evaluation in the correct order to work safely. I am not sure whether this will be given for all compilers and it is not very readable what happens here. Can we make this into an if statement ( we could use the : operator for this)?

Reply:
That is (luckily!) one thing we really can rely on compilers to implement properly. We use this in multiple of places:
```
$ grep -r packet_version src/ | grep '&&'
src/economy/expedition_bootstrap.cc:			bool removable = (packet_version >= 8) && (fr.unsigned_8() != 0);
src/map_io/map_flagdata_packet.cc:						flag.act_pending_ = packet_version >= 6 && (fr.unsigned_8() != 0u);
src/map_io/map_buildingdata_packet.cc:					building.mute_messages_ = packet_version >= 6 && (fr.unsigned_8() != 0u);
src/map_io/map_buildingdata_packet.cc:					building.is_destruction_blocked_ = (packet_version >= 9) && (fr.unsigned_8() != 0);
src/map_io/map_buildingdata_packet.cc:			productionsite.infinite_production_ = packet_version >= 10 && fr.unsigned_8() > 0;
```
Actually if we used an if or ternary here clang-tidy would tell us just to logically concatenate the bool expressions instead.